server/adaptors/ppl_plugin.ts

```diff
@@ -37,4 +41,17 @@ export const PPLPlugin = function (Client, config, components) {
     needBody: true,
     method: 'POST',
   });
+
+  ppl.datasourceQuery = ca({
+    url: {
+      fmt: `${OPENSEARCH_DATASOURCES_API.DATASOURCE}/<%=datasource%>`,
+      req: {
+        datasource: {
+          type: 'string',
+          required: false,
+        },
+      },
+    },
+    method: 'GET',
+  });
```

Comment:
I don't know if this is the correct way to optionally include datasource, or if it is even a good practice to share a same client action for two different calls, but it works locally

Reply:
Done [bc88ebe](https://github.com/opensearch-project/dashboards-observability/pull/967/commits/bc88ebe4cc2191a721a8e9bd36f0e57ef8e157b5)